### PR TITLE
Switch to os js sdk

### PR DESCRIPTION
**Pull request by jjtang1985**
- [x] switch sdk dependencies
- [x] fix the breaking changes that happens on the cloud
- [x] switch from 2019 to 2020
- [x] fix broken links

### CODE REVIEW

**Comment by jjtang1985** [comment]
@marikaner , please review.

File: exercises/01-getting-started.md [hunk 1]
```diff
@@ -2,8 +2,6 @@
 In the following we describe how to get started with this project.
 You will create a new **private** repository based on this repository and run the application locally to get familiar with it.
 
-> In the session, you will be handed out a participantId. If you go through the tasks on your own, make up your own id. We recommend to refrain from using personal data.
```
Comment (marikaner) [inline exercises/01-getting-started.md:5]: I think this note might be helpful. How about rewriting it to something like this: 

> In the original session, the participants were handed out a participantId. If you go through the tasks on your own, make up your own id. We recommend to refrain from using personal data.

File: exercises/04-deploy-to-cf.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ We will need some services for deployment to Cloud Foundry. We will need an HDI
 ### HDI Container
 In this application we are storing / reading data from SAP S/4HANA, SAP SuccessFactors and a local database. Create an HDI container to store local data in. Don't forget to replace the participantId:
 ```sh
-cf create-service hanatrial hdi-shared timesheet-hdi-container-codejam-<participantId>
+cf create-service hanatrial hdi-shared cloud-sdk-team-calendar-db-hdi-container
```
Comment (marikaner) [inline exercises/04-deploy-to-cf.md:19]: I think it is better if everyone has their own container using the participantId. You mentioned, that using the same name could be an issue. Why changing it then?

File: exercises/04-deploy-to-cf.md [hunk 3]
```diff
@@ -48,7 +48,12 @@ Run the following to build and package your application:
 npm run ci-build && npm run ci-package
 ```
 
-Then push it to Cloud Foundry:
+Then deploy the data defined in the CSVs to the HDI container:
```
Comment (marikaner) [inline exercises/04-deploy-to-cf.md:51]: How is this deploying anything to the HDI container? We ran this in the getting started section. The documentation there states, that this deploys the database locally. How can it now do something else?

If I went through this sample I would not understand the difference between what we did in the getting started section and here.

File: exercises/07-automate-deployment.md [hunk 4]
```diff
@@ -38,7 +38,6 @@ Those files are necessary to run the pipeline with Travis CI:
 
 ## Adjust pipeline configuration
 Go to the [pipeline_config.yml](pipeline_config.yml) file and adjust it to your project. Set the org and space for the `productionDeployment` as you have done before when logging in to Cloud Foundry (if you don't remember run `cf target` on the command line to get a summary).
-Also, replace your participantId for the `appName`.
```
Comment (marikaner) [inline exercises/07-automate-deployment.md:41]: Why did you remove this sentence? It could be slightly improved, but I think this is very relevant, so nobody forgets.

Suggestion: Also, replace your participantId in the `appName`.

Comment (marikaner) [reply exercises/07-automate-deployment.md:41]: One more thing: I realized that most of the links on this page (and potentially others) are incorrect: [pipeline_config.yml](pipeline_config.yml) should be [pipeline_config.yml](../pipeline_config.yml).
I created a ticket for that, but if you can, maybe you can make this a part of this PR already.

File: manifest.yml [hunk 5]
```diff
@@ -1,12 +1,12 @@
 applications:
-  - name: timesheet-codejam-<participantId>
+  - name: cloud-sdk-team-calendar-db-hdi-container
     path: deployment/
     buildpacks:
       - nodejs_buildpack
     memory: 256M
-    command: node node_modules/@sap/hdi-deploy/deploy.js -t --verbose --root ./db & node dist/index.js
+    command: node dist/index.js
```
Comment (marikaner) [inline manifest.yml:7]: 👍

File: exercises/04-deploy-to-cf.md (hunk 3 above)
Comment (marikaner) [reply exercises/04-deploy-to-cf.md:51]: Edit: I now scrolled further down and saw, that you changed the `npm cds-deploy` script. Consider putting this into the manifest.yml or alternatively create two scripts, one for local, one for productive deployment (e. g. `npm cds-deploy:local` and `...:prod`). Don't forget to mention what the difference is.

File: manifest.yml [hunk 6]
```diff
@@ -1,12 +1,12 @@
 applications:
-  - name: timesheet-codejam-<participantId>
+  - name: cloud-sdk-team-calendar-db-hdi-container
```
Comment (marikaner) [inline manifest.yml:2]: The application should not use the same name as the HDI container. What was wrong with the old name? You may want to remove the notion of codejam and use something like sample instead.

File: package.json [hunk 7]
```diff
@@ -7,7 +7,8 @@
     "init": "npm i && node initialize.js",
     "preinstall": "rm -rf node_modules/@sap/cds-services",
     "cds-build": "npx cds build/all --clean",
-    "cds-deploy": "npx cds deploy",
+    "cds-deploy:local": "npx cds deploy",
+    "cds-deploy:hana": "npx cds deploy --to hana:timesheet-hdi-container-codejam-<participantId>",
```
Comment (jjtang1985) [inline package.json:11]: here is how you can customise the container name

File: exercises/04-deploy-to-cf.md [hunk 8]
```diff
@@ -38,17 +38,22 @@ This is what the service instances should look like in the SAP Cloud Platform co
...
 ## Build and push your application
 Run the following to build and package your application:
 ```sh
 npm run ci-build && npm run ci-package
 ```
 
-Then push it to Cloud Foundry:
+Then, open the [package.json](../package.json), replace the `<participantId>` in the script `cds-deploy:hana` and deploy the data defined in the CSVs to the HDI container:
+```sh
+npm run cds-deploy:hana
```
Comment (marikaner) [inline exercises/04-deploy-to-cf.md:53]: Minor: It might be slightly less confusing to just add this command to the manifest.yml (run `npm run cds-deploy:hana && node dist/index.js`) and just mention what this does instead of having everyone doing this manually.

Comment (jjtang1985) [reply exercises/04-deploy-to-cf.md:53]: good point, unfortunately, it will result in the error below:
```
2020-04-09T14:14:14.980+0000 [APP/PROC/WEB/0] OUT [cds.deploy] - Starting deploy to SAP HANA ...
2020-04-09T14:14:14.981+0000 [APP/PROC/WEB/0] OUT [cds.deploy] - Creating build tasks
2020-04-09T14:14:14.981+0000 [APP/PROC/WEB/0] OUT [cds.deploy] - Running build
2020-04-09T14:14:16.880+0000 [APP/PROC/WEB/0] OUT [cds.deploy] - Running 'cf marketplace -s hana' with options {}
2020-04-09T14:14:16.888+0000 [APP/PROC/WEB/0] ERR [ERROR] Command 'cf' not found. Make sure you have the Cloud Foundry command line tool installed.
2020-04-09T14:14:16.888+0000 [APP/PROC/WEB/0] ERR     at Function._getError (/home/vcap/deps/0/node_modules/@sap/cds/bin/deploy/to-hana/cfUtil.js:149:20)
2020-04-09T14:14:16.888+0000 [APP/PROC/WEB/0] ERR     at Function.hasMarketplaceService (/home/vcap/deps/0/node_modules/@sap/cds/bin/deploy/to-hana/cfUtil.js:22:24)
2020-04-09T14:14:16.888+0000 [APP/PROC/WEB/0] ERR     at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-04-09T14:14:17.002+0000 [APP/PROC/WEB/0] OUT Exit status 1
```
The `cf` command is then needed for the cds-deploy:hana.
I guess this is not designed for running on the cloud.

Comment (jjtang1985) [reply exercises/04-deploy-to-cf.md:53]: For this question, we can open another issue to clarify.

Comment (marikaner) [reply exercises/04-deploy-to-cf.md:53]: Thanks for checking that. This was rather an idea than a necessity. It is fine as is 😊
